src/microbe_stage/components/Engulfable.cs

```diff
@@ -295,7 +295,8 @@ public static void OnReportBecomeIngestedIfCallbackRegistered(this ref Engulfabl
 
                     MicrobeDeathSystem.SpawnCorpseChunks(ref organelleContainer,
                         entity.Get<CompoundStorage>().Compounds, spawnSystem, worldSimulation, recorder,
-                        position.Position, new XoShiRo128starstar(), customizeCallback, null);
+                        position.Position, new XoShiRo128starstar(), customizeCallback, null, entity
+                            .Get<MicrobeSpeciesMember>().Species.IsBacteria);
```

Comment:
This should access `CellProperties` to check if the cell is bacteria. Accessing the species data is incorrect here as it might not match the spawned cell.

Reply:
This method already accesses CellProperties below, but due to ordering of the logic you probably need to add a new block that checks if the component exists (and caches that as a bool) and then stores the fact that if the cell is bacteria or not in a variable that is then read on this line.